src/luarocks/deps.lua

```diff
@@ -703,6 +703,9 @@ local function lua_h_exists(d, luaver)
       if data:match("LUA_VERSION_NUM%s*" .. tostring(luanum)) then
          return d
       end
+      if data:match("LUA_VERSION_MINOR_N%s*" .. tostring(minor)) then
```

Comment:
Should this also check `LUA_VERSION_MAJOR_N`, so that it's both equivalent and Lua-6-ready?

Reply:
A *lot* of other things would break in LuaRocks with a hypothetical Lua 6. Actually, even for Lua 5.5 it would require manual intervention to make LuaRocks accept the new version.

Given Lua's evolution history, LuaRocks is not written using the assumption than a 5.(x+1) would ever work out-of-the-box. There were changes in past Lua versions which included standard library functions remaining syntactically compatible but semantically changed. For this reason, we actually prevent an unknown version from running, to avoid having it "almost work" by accident and then fail in insidious ways.

Reply:
@hishamhm Understood, but that's sort of my point.

In the incredibly unlikely event that Lua's release cadence massively accelerates, **in _theory_** if this check were to encounter a future Lua 6.4 release, it would accept it under the assumption that it's actually Lua 5.4.

Reply:
(When I say "Lua 6 ready", I don't mean "Lua 6 compatible" so much I mean ready to _reject_ Lua 6.x releases as **incompatible**. Right now, the new checks can't differentiate between Lua 6.1 and Lua 5.1, or Lua 7.2 and Lua 5.2.

Reply:
Fair point!